pkg/kubectl/cmd/rollingupdate.go

```diff
@@ -283,3 +292,21 @@ func findNewName(args []string, oldRc *api.ReplicationController) string {
 	}
 	return ""
 }
+
+func isReplicasDefaulted(info *resource.Info) bool {
+	if info == nil || info.VersionedObject == nil {
+		// was unable to recover versioned info
+		return false
+	}
+	switch info.Mapping.APIVersion {
+	case "v1":
+		if rc, ok := info.VersionedObject.(*v1.ReplicationController); ok {
```

Comment:
I still don't think you need a VersionedObject. If the return case is the same for v1beta3 and v1, why not just
```
if rc, ok := info.Object.(api.ReplicationController); ok {
			return rc.Spec.Replicas == nil
}
```
Is there another reason for adding a VersionedObject besides this particular use?